store/src/db/sql.rs

```diff
@@ -153,28 +157,51 @@ pub fn select_account(
 /// transaction.
 pub fn upsert_accounts(
     transaction: &Transaction,
-    accounts: &[(AccountId, Option<AccountDetails>, RpoDigest)],
+    accounts: &[AccountDetailsUpdate],
     block_num: BlockNumber,
 ) -> Result<usize> {
-    let mut stmt = transaction.prepare(
-        "
-        INSERT OR REPLACE INTO
-            accounts (account_id, account_hash, block_num, details)
-        VALUES (?1, ?2, ?3, ?4);
-    ",
+    let mut upsert_stmt = transaction.prepare(
+        "INSERT OR REPLACE INTO accounts (account_id, account_hash, block_num, details) VALUES (?1, ?2, ?3, ?4);",
     )?;
+    let mut select_details_stmt =
+        transaction.prepare("SELECT details FROM accounts WHERE account_id = ?1;")?;
 
     let mut count = 0;
-    for (account_id, details, account_hash) in accounts.iter() {
-        // TODO: Process account details/delta (in the next PR)
+    for update in accounts.iter() {
+        let account_id = update.account_id.into();
+        let full_account = match &update.details {
+            None => None,
+            Some(AccountDetails::Full(account)) => {
+                debug_assert!(account.is_new());
+                debug_assert_eq!(account_id, u64::from(account.id()));
+
+                Some(Cow::Borrowed(account))
+            },
```

Comment:
Should we not check here if final account hash matches the account hash from details and return an error if not?